src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -218,47 +317,49 @@ pub fn update_lockfile(ws: &Workspace<'_>, opts: &UpdateOptions<'_>) -> CargoRes
     fn compare_dependency_graphs(
```

Comment:
nit: could we have a comment or return a struct so that people can look at it without read through the entire implementation to understand what it returns?